File: LeetCode/cpp/lc_0027.cpp
/**
 * @file     lc_0027.cpp
 * @brief    https://leetcode-cn.com/problems/remove-element/
 * @author   YongDu
 * @date     2021-09-05
 */
class Solution {
  public:
    int removeElement(vector<int>& nums, int val) {
        int len = 0;
        for (int n : nums) {
            if (n != val) {
                nums[len++] = n;
            }
        }
        return len;
    }
};

File: LeetCode/python/lc_0027.py
"""
 * @file     removeElement.py
 * @brief    https://leetcode-cn.com/problems/remove-element/
 * @author   YongDu
 * @date     2021-08-18
"""


class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        index = 0
        for num in nums:
            if num != val:
                nums[index] = num
                index += 1
        return index
